components/cosmetic_filters/renderer/cosmetic_filters_js_render_frame_observer.cc

```diff
@@ -78,30 +77,23 @@ void CosmeticFiltersJsRenderFrameObserver::ReadyToCommitNavigation(
   if (!url_.SchemeIsHTTPOrHTTPS())
     return;
 
-  native_javascript_handle_->ProcessURL(
-      url_, base::BindOnce(&CosmeticFiltersJsRenderFrameObserver::OnProcessURL,
-                           weak_factory_.GetWeakPtr()));
+  bool continue_with_cf = native_javascript_handle_->ProcessURL( url_);
+
+  if (continue_with_cf) {
```

Comment:
nit: what is the purpose of `continue_with_cf`? why not just `should_inject_ = native...`?

Reply:
Good call; `should_inject_` was my replacement for the `OneShotSignal` we had here before and I just did a 1:1 update here without simplifying the logic.